web-ui/src/main/resources/catalog/components/common/map/mapsManager.js

```diff
@@ -182,10 +182,16 @@
             if (config.context) {
               mapReady = gnOwsContextService.loadContextFromUrl(
                 config.context, map);
+            } else {
+              //Failback needed for controllers that rely on map creation
+              mapReady = $q.defer();
+              mapReady.resolve();
+              mapReady = mapReady.promise;
             }
           }
-          var creationPromise = $q.when(mapReady).then(function() {
-
+          
+          //This should be called also when resetting the default map
+          $rootScope.$on('owsContextLoaded', function() {
```

Comment:
Dummy promise, but makes easier to handle this function the same way as if it was a loadContextByUrl.
Moved event to the lower end of the function, which is when the context has finished loading

Reply:
As we now use the event to add the layers from settings, now if the context is loaded anywhere in the code, the layers from settings will follow.